src/UXClient/Components/LineChart/LineChart.ts

```diff
@@ -1286,9 +1288,116 @@ class LineChart extends TemporalXAxisComponent {
+            }
+        }
+
+        // First add numeric dataTypes (share Y-Axis) to label map
+        visibleCDOs.filter(aggGroup => aggGroup.dataType === DataTypes.Numeric).forEach(aggGroup => {
+            if(!(aggGroup.swimLane in swimLaneLabels)){ // Only add swimlanes once to swimLaneLabels map
+                useAggForLaneLabel(aggGroup);
+            }
+        })
+
+        // Then, map over any non-numeric dataType and increment heights if they're sharing a lane
+        visibleCDOs.filter(aggGroup => aggGroup.dataType !== DataTypes.Numeric).forEach(aggGroup => {
+            let aggIndex = visibleCDOs.findIndex(el => el.aggKey === aggGroup.aggKey);
+            if(!(aggGroup.swimLane in swimLaneLabels)){ // Only add swimlanes once to swimLaneLabels map
+                useAggForLaneLabel(aggGroup);
+            } else{ // if lane contains non-numeric data and is being added to another lane
+                if(!this.chartOptions?.swimLaneOptions?.[aggGroup.swimLane]?.collapseEvents){ // Only increment event heights if collapseEvents === false
+                    swimLaneLabels[aggGroup.swimLane].height += offsetsAndHeights[aggIndex][1]; // add heights (non-numerics don't share Y axis)
+                }
+            }
+        });
+
+        // Clear prior labels
+        this.swimLaneLabelGroup.selectAll('*').remove();
+
+        // Function to trim labels to max height
+        const truncateLabel = (labelRef: HTMLElement, data) => {
+            const maxHeight = data.height - this.swimLaneLabelHeightPadding; // 8px padding on actual lane height
+            if(data.label){
+                let labelClientRect = labelRef.getBoundingClientRect();
+                let labelText = labelRef.textContent;
+
+                while ( labelClientRect.height > maxHeight && labelText.length > 0) {
+                    labelText = labelText.slice(0, -1);
+                    labelRef.textContent = labelText + '...';
+                    labelClientRect = labelRef.getBoundingClientRect();
+                }
+            }
+        }
+
+        const boldYAxisText = (enabled: boolean, lane: string) => {
+            this.svgSelection.select('.svgGroup')
+                .selectAll(`.tsi-swimLaneAxis-${lane}`)
+                .selectAll('text')
+                .classed('boldYAxisText', enabled)
+        }
+
+        // Map over swimLanes and create labels
+        Object.keys(swimLaneLabels).forEach(lane => {
+            let labelData = [swimLaneLabels[lane]];
+            let label = this.swimLaneLabelGroup.selectAll(`tsi-swimLaneLabel-${lane}`).data(labelData);
+
+            label.enter()
+                .append("text")
+                .attr("class", `tsi-swimLaneLabel-${lane} tsi-swimLaneLabel`)
+                .merge(label)
+                .style("text-anchor", "middle")
+                .attr("transform", d => `translate(${( -this.horizontalLabelOffset + 28 )},${(d.offset + d.height / 2)}) rotate(-90)`)
```

Comment:
I don't know, in general should we have constants for all these kind of values like 28px? What do you think?

Reply:
Agreed!  I'll pull in these constants from the `Constants.ts` file